crates/shared/src/order_validation.rs

```diff
@@ -669,16 +655,37 @@ impl OrderValidating for OrderValidator {
                         buy: quote.buy_amount,
                         fee: quote.fee_amount,
                     },
-                ) =>
-            {
-                tracing::debug!(%uid, ?owner, ?class, "order being flagged as outside market price");
-                OrderClass::Liquidity
+                ) {
+                    tracing::debug!(%uid, ?owner, ?class, "order being flagged as outside market price");
+                    (OrderClass::Liquidity, Some(quote))
+                } else {
+                    (class, Some(quote))
+                }
+            }
+            OrderClass::Limit => {
+                let quote =
+                    get_quote_and_check_fee(&*self.quoter, &quote_parameters, order.quote_id, None)
+                        .await?;
+                // If the order is not "In-Market", check for the limit orders
+                if is_order_outside_market_price(
+                    &Amounts {
+                        sell: data.sell_amount,
+                        buy: data.buy_amount,
+                        fee: data.fee_amount,
+                    },
+                    &Amounts {
+                        sell: quote.sell_amount,
+                        buy: quote.buy_amount,
+                        fee: quote.fee_amount,
+                    },
+                ) {
+                    self.check_max_limit_orders(owner).await?;
+                }
+                (class, Some(quote))
             }
-            (_, _) => class,
+            OrderClass::Liquidity => (class, None),
         };
 
-        self.check_max_limit_orders(owner, &class).await?;
```

Comment:
Why not just substituting this with:

```
      if class.is_limit() { // to be deleted after market/liquidity order cleanup
            let quote = quote.unwrap(); // to be deleted after market/liquidity order cleanup
            if is_order_outside_market_price(
                &Amounts {
                    sell: data.sell_amount,
                    buy: data.buy_amount,
                    fee: data.fee_amount,
                },
                &Amounts {
                    sell: quote.sell_amount,
                    buy: quote.buy_amount,
                    fee: quote.fee_amount,
                },
            ) {
                self.check_max_limit_orders(owner).await?;
            }
        }
```

Proper refactoring would be done once the cleanups for market and liquidity orders happen.

Reply:
That is a good point, but I did it with the match because:
- we can check if the market order is limit or market (legacy, this should be deleted once we only have limit)
- map the liquidity orders to limit orders (so we eliminate the liquidity orders)
it should be quite easy to reduce this code once we remove liquidity and market orders completely.